pkg/jiraservicedesk/client/client_test.go

```diff
@@ -0,0 +1,115 @@
+package client
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/nbio/st"
+	"gopkg.in/h2non/gock.v1"
+)
+
+const BaseURL = "https://sample.atlassian.net/"
+
+func TestJiraService_GetProject_shouldGetProjectById(t *testing.T) {
+	defer gock.Off()
+
+	// The project.LeadAccountId & project.ProjectTemplateKey are not returned by the GET call. Therefore, those values are not returned and matched via Gock Server
+	gock.New(BaseURL + "rest/api/3/project").
+		Get("/10003").
+		Reply(200).
+		JSON(map[string]string{
+			"description":    "Sample Project",
+			"name":           "Sample",
+			"assigneeType":   "UNASSIGNED",
+			"projectTypeKey": "business",
+			"key":            "KEY",
+			"url":            "https://www.sample.com",
+		})
+
+	jiraClient := NewClient("", BaseURL, "")
+	project, err := jiraClient.GetProjectById("/10003")
+
+	st.Expect(t, project.Description, "Sample Project")
+	st.Expect(t, project.Name, "Sample")
+	st.Expect(t, project.AssigneeType, "UNASSIGNED")
+	st.Expect(t, project.ProjectTypeKey, "business")
+	st.Expect(t, project.Key, "KEY")
+	st.Expect(t, project.URL, "https://www.sample.com")
+	st.Expect(t, err, nil)
+
+	st.Expect(t, gock.IsDone(), true)
+}
+
+func TestJiraService_CreateProject_shouldCreateProject(t *testing.T) {
+	defer gock.Off()
+
+	gock.New(BaseURL + "/rest/api/3/project").
+		Post("/").
+		MatchType("json").
+		JSON(map[string]string{
+			"description":        "Sample Project",
+			"leadAccountId":      "5ebfbc3ead226b0ba46c3591",
+			"projectTemplateKey": "com.atlassian.jira.jira-incident-management-plugin:im-incident-management",
+			"name":               "Sample",
+			"assigneeType":       "UNASSIGNED",
+			"projectTypeKey":     "business",
+			"key":                "KEY",
+		}).
+		Reply(200).
+		JSON(map[string]interface{}{"self": "https://jira-service-desk.net/rest/api/3/project/10003", "id": 10003, "key": "KEY"})
```

Comment:
Base URL in self link URL differs from the one you defined at https://github.com/stakater/jira-service-desk-operator/pull/24/files#diff-05c8f5c5057717969a56ba3950662f49R12